components/feature/push/src/main/java/mozilla/components/feature/push/ext/Connection.kt

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.push.ext
+
+import mozilla.components.feature.push.PushConnection
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Executes the block if the Push Manager is initialized.
+ */
+internal inline fun PushConnection.ifInitialized(block: PushConnection.() -> Unit) {
+    if (isInitialized()) {
+        block()
+    } else {
+        Logger.error("Native push layer is not yet initialized.")
```

Comment:
do we never need to hard-assert this condition and crash out?

Reply:
We spent some time thinking about this: push fatal errors are limited to those are not in this list: https://github.com/mozilla-mobile/android-components/pull/6397/files#diff-c2cb12518c4508eaf1fb172996e64fd5R328

The fatal errors we can receive are malformed messages or unrecoverable database errors. Fatal for push doesn't mean we should crash out for the user because we have no real recovery process aside using `verifyConnection` on the AutoPush side and `renewRegistration` for resetting our push tokens.

So for now, the best we can do is report those fatal errors to Sentry and fix them in future releases. App services, could always handle or expose a way to reset the internal db, but that's out of the scope right now until we know what are the situations we want to handle for that solution.

Reply:
This was our way of avoiding any timing issues where a message or token could be received from the push service _before_ our rust code had initialized. This shouldn't happen in practice since we don't allow receiving messages until everything is initialized first - it ends up being a way to satisfy the compiler checks that what could happen in practice.

If we lose:
 - A message, this would a lost push message that we cannot recover. In theory we could create a buffer to process these messages layer but in the worse case this would be only the first message loss (see above why this wouldn't happen).
- A push token, we still store the token in our prefs and re-init our native layer with it on the next initialization.